lib/gemini.js

```diff
@@ -23,39 +26,49 @@ function requireWithNoCache(moduleName) {
     return result;
 }
 
-function Gemini(config, overrides) {
-    config = config || DEFAULT_CFG_NAME;
-    this.config = new Config(config, overrides);
-    var _this = this;
+module.exports = inherit(EventEmitter, {
+    __constructor: function(config, overrides) {
+      config = config || DEFAULT_CFG_NAME;
```

Comment:
Here's few ways I achieved the running order I was looking for. They're both ugly, do you know if there is a way to do the same thing with some built-in method found in Q/Promise ?

Nested:
```
          return _this.readTests(paths)
            .then(function(rootSuite) {
              return _this.emit('startRunner', runnerInstance)
                .then(function () {
                  options.reporters.forEach(applyReporter.bind(null, runnerInstance));
                  return runnerInstance.run(rootSuite);
                });
              })
            .then(function(data) {
                return _this.emit('endRunner', runnerInstance, data)
                  .then(function() {
                    return data;
                  });
            });
```

Using temp vars:
```
return _this.readTests(paths)
              .then(function(rootSuite) {
                  _this.rootSuite = rootSuite;
              })
              .then(function() {
                  return _this.emit('startRunner', runnerInstance);
              })
              .then(function() {
                  options.reporters.forEach(applyReporter.bind(null, runnerInstance));
                  return runnerInstance.run(_this.rootSuite);
              })
              .then(function(data) {
                  _this.data = data;
                  return _this.emit('endRunner', runnerInstance, data);
              })
              .then(function() {
                  return _this.data;
              });
```